Userland/Libraries/LibTest/Randomized/Generator.h

```diff
@@ -0,0 +1,328 @@
+/*
+ * Copyright (c) 2023, Martin Janiczek <martin@janiczek.cz>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <LibTest/Macros.h>
+#include <LibTest/Randomized/RandomRun.h>
+
+#include <AK/Function.h>
+#include <AK/Random.h>
+#include <AK/String.h>
+#include <AK/StringView.h>
+#include <AK/Tuple.h>
+
+namespace Test {
+namespace Randomized {
+
+// Returns a random double value in range 0..1.
+inline double get_random_probability()
+{
+    static constexpr u32 max_u32 = NumericLimits<u32>::max();
+    u32 random_u32 = AK::get_random_uniform(max_u32);
+    return static_cast<double>(random_u32) / static_cast<double>(max_u32);
+}
+
+// Generators take random bits from the RandomnessSource and return a value
+// back.
+//
+// Example:
+// - Gen::u32(5,10) --> 9, 7, 5, 10, 8, ...
+namespace Gen {
+
+// An unsigned integer generator.
+//
+// The minimum value will always be 0.
+// The maximum value is given by user in the argument.
+//
+// Gen::unsigned_int(10) -> value 5, RandomRun [5]
+//                       -> value 8, RandomRun [8]
+//                       etc.
+//
+// Shrinks towards 0.
+inline u32 unsigned_int(u32 max)
+{
+    u32 random = Test::randomness_source().draw_value(max, [&]() {
+        return AK::get_random_uniform(max + 1);
+    });
+    return random;
+}
+
+// An unsigned integer generator in a particular range.
+//
+// The minimum value is the smaller of the two arguments.
+// The maximum value is the larger of the two arguments.
+//
+// Gen::unsigned_int(3,10) -> value 3,  RandomRun [0]
+//                         -> value 8,  RandomRun [5]
+//                         -> value 10, RandomRun [7]
+//                         etc.
+//
+// In case `min == max`, the RandomRun footprint will be smaller, as we'll
+// switch to a `constant` and won't need any randomness to generate that
+// value:
+//
+// Gen::unsigned_int(3,3) -> value 3, RandomRun [] (always)
+//
+// Shrinks towards the smaller argument.
+inline u32 unsigned_int(u32 min, u32 max)
+{
+    if (min > max) {
+        return unsigned_int(max, min);
+    }
```

Comment:
`VERIFY` inside tests is redefined and doesn't abort. It would only mark the test as failed. Maybe we should alias `VERIFY` to some other name in `Macros.h` to keep access to its original behaviour, before we redefine it?

Reply:
Oh, right, that's a thing. Ugh.

Maybe we want to get rid of the whole "we know that something is wrong, lets run into undefined behavior anyways" instead. But yeah, for now, renaming the macro (and maybe adding a `FIXME` seems like the best option.

Reply:
For posterity: we are removing the VERIFY redefinition so that (among other things) we can use VERIFY here and abort.
https://github.com/SerenityOS/serenity/pull/21479